src/main/java/org/example/Phone.java

```diff
@@ -0,0 +1,144 @@
+package org.example;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This is a class that implements the  PhoneInterface. It contains the methods that are define in the interface.
+ *
+ * @param : -  protected final Integer batteryLife;
+ * @param : -  List<Contact> contact = new ArrayList<>();
+ * @param : - Map<String, List<String>> messages = new HashMap<>();
+ * @param : - List<String> calls =new ArrayList<>();
+ * @param : - Integer battery ;
+ */
+abstract class Phone implements PhoneInterface {
+    protected final Integer batteryLife;
+    List<Contact> contact = new ArrayList<>();
+    Map<String, List<String>> messages = new HashMap<>();
+    List<String> calls = new ArrayList<>();
+    Integer battery;
+
+    /**
+     * This is the constructor of the class Phone.
+     *
+     * @param : -  batteryLife;
+     */
+    Phone(Integer batteryLife) {
+
+        this.batteryLife = batteryLife;
+        battery = batteryLife;
+
+    }
+
+    /**
+     * This is a method that add a contact to the list.
+     *
+     * @param : -  name;
+     * @param : - phoneNumber;
+     */
+    @Override
+    public void addContact(String name, String phoneNumber) {
+        Contact personFromList = new Contact(name, phoneNumber);
+        contact.add(personFromList);
+        System.out.println(personFromList);
+
+    }
+
+    /**
+     * This is a method that get a contact from the list.
+     *
+     * @param : -  index;
+     */
+    public Contact getContact(int index) {
+        if (index >= 0 && index < contact.size()) {
+            contact.get(index);
+        } else {
+            System.out.println("Index out of range ");
+        }
+        return contact.get(index);
+    }
+
+    /**
+     * This is a method that return the phone number.
+     *
+     * @param : -i- the index;
+     */
+    public String getNumber(int i) {
+        ;
+        return contact.get(i).phoneNumber;
+    }
+
+    /**
+     * This is a method that put the message in a HashMap.
+     *
+     * @param : - phoneNumber;
+     * @param : - message;
+     */
+    public void sendMessage(String phoneNumber, String message) {
+        if (message.length() <= 500) {
+            for (int i = 0; i < contact.size(); i++) {
+                if (getNumber(i) == phoneNumber) {
+                    if (!messages.containsKey(phoneNumber)) {
+                        messages.put(phoneNumber, new ArrayList<>());
+                    }
+                    messages.get(phoneNumber).add(message);
+                    System.out.println("Message " + "'" + message + "'"
+                            + " is sent to phone number " + phoneNumber);
+
+                    battery -= 1;
+                }
+            }
+        } else {
```

Comment:
you can transform this into a guard condition, if (message>500), error and exit method